lib/Parser/OglParser.php

```diff
@@ -23,32 +23,34 @@
 class OglParser
 {
     private $hints;
+    private $matrix;
 
-    public function __construct($hints) {
+    public function __construct($hints, $matrix) {
         $this->hints = $hints;
+        $this->matrix = $matrix;
     }
 
-    public function parse($filename) {
+    public function parse($filename, $commit = null) {
         $handle = fopen($filename, "r");
```

Comment:
Maybe it's because I'm more used to C++, but here is my point:

Let's say we have this line:

`parse($filename, $commit);`

Then you change the method to:

`public function parse($filename, $important_param, $commit = null) { ...`

Because of the default parameter, we might even never see that there is an
error here. Since `$commit` is a class, I admit that the possibility to
have no error at all, even during execution, is quite limited. But the
principle stays the same: without default parameters, this case never
happens.

As I said, that's maybe because I too used to compiled languages. But I do
think this problem applies to all languages actually. I'm not the only one
thinking that (I'm not crazy! :)). You can look Johnathan Blow's videos for
instance.